src/codegen.cpp

```diff
@@ -4612,6 +4634,22 @@ static void undef_var_error_ifnot(jl_codectx_t &ctx, Value *ok, jl_sym_t *name)
     ctx.builder.SetInsertPoint(ifok);
 }
 
+static void emit_hasnofield_error_ifnot(jl_codectx_t &ctx, Value *ok, jl_datatype_t *type, jl_cgval_t name)
```

Comment:
`type` here is not quite rooted. You probably just want `emit_typeof(obj)`

Reply:
or pass `type->name` here, which is rooted

Reply:
I thought types are always rooted, but will do the change anyway